### ISSUE TITLE
customAutoEscapedFunctions array property not recognised in new format

### ISSUE THREAD

**Issue by GaryJones** [WordPress-Coding-Standards/WordPress-Coding-Standards#1504]
PHPCS: 3.3.2
WPCS: 1.1

PHPCS 3.3 introduced a new format for setting array properties. I've used this successfully for one or more i18n text domains, one or more prefixes and so on. All good.

When trying the same with `customAutoEscapedFunctions`, I'm getting violations appear with the attempted new format (as though I hadn't set it), that didn't appear with the old format.

Works (violations do not appear):

```xml
<rule ref="WordPress.Security.EscapeOutput">
	<properties>
		<property name="customAutoEscapedFunctions" type="array" value="genesis_attr"/>
	</properties>
</rule>
```

Doesn't work:

```xml
<rule ref="WordPress.Security.EscapeOutput">
	<properties>
		<property name="customAutoEscapedFunctions" type="array">
			<element name="genesis_attr"/>
		</property>
	</properties>
</rule>
```

At a _guess_, some logic around [here](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/1cb6a74615be51390776c3c22438c21b010c1737/WordPress/Sniffs/Security/EscapeOutputSniff.php#L524) is not accounting for the new format?

**Comment by jrfnl** [comment]
@GaryJones Without having done any tests, I'm reasonably confident it's not a WPCS code issue.

Try this and take note of `value` not `name` in the `element`:
```xml
<rule ref="WordPress.Security.EscapeOutput">
	<properties>
		<property name="customAutoEscapedFunctions" type="array">
			<element value="genesis_attr"/>
		</property>
	</properties>
</rule>
```

See: https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.3.0

**Comment by GaryJones** [comment]
Dammit - I even looked at that but didn't actually check it. My boilerplate code has it down as `name`. Maybe for typical instances it doesn't matter which is used, but for this comparison, it does?

**Comment by jrfnl** [comment]
In that case, I suggest you start using the XSD file to validate your rulesets :smirk: 

See: https://github.com/squizlabs/PHP_CodeSniffer/pull/1433

Note: the new array format is only included in the XSD file since PHPCS 3.3.2.
